MediaContentSynchronizationApi/Model/SynchronizerPool.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentSynchronizationApi\Model;
+
+use Magento\MediaContentSynchronizationApi\Api\SynchronizeInterface;
+
+/**
+ * A pool of content with assets synchronizers
+ * @see SynchronizeFilesInterface
+ */
+class SynchronizerPool
+{
+    /**
+     * Content with assets synchronizers
+     *
+     * @var SynchronizeInterface[]
+     */
+    private $synchronizers;
+
+    /**
+     * @param SynchronizeInterface[] $synchronizers
+     */
+    public function __construct(
+        array $synchronizers = []
+    ) {
+        foreach ($synchronizers as $synchronizer) {
+            if (!$synchronizer instanceof SynchronizeInterface) {
+                throw new \InvalidArgumentException(__('Synchronizer doesn\'t implement SynchronizeInterface'));
```

Comment:
Can you please also add this check to MediaGallery synchronizer pool for consistency